network/jump.go

```diff
@@ -15,30 +15,12 @@ var _ Dialer = (*JumpDialer)(nil)
 // JumpDialer is a RetryDialer from a proxy.
 type JumpDialer struct {
 	RetryDialer
-	user    string
-	addr    string
-	keyfile string
+	client *ssh.Client
 }
 
 // NewJumpDialer initializes a JumpDialer to establish a SSH Proxy Jump.
-func NewJumpDialer(addr, user, keyfile string) *JumpDialer {
-	return &JumpDialer{
-		RetryDialer: RetryDialer{
-			Dialer: &net.Dialer{
-				Timeout:   DefaultTimeout,
-				KeepAlive: DefaultKeepAlive,
-			},
-			Retries: DefaultRetries,
-		},
-		user:    user,
-		addr:    addr,
-		keyfile: keyfile,
-	}
-}
-
-// Dial connects to a remote address, retrying on failure.
-func (d *JumpDialer) Dial(network, address string) (c net.Conn, err error) {
-	key, err := ioutil.ReadFile(d.keyfile)
+func NewJumpDialer(addr, user, keyfile string) (*JumpDialer, error) {
```

Comment:
Following the suggestion below, then do this
```suggestion
func NewJumpDialer(addr, user, keyfile string) (*RetryDialer, error) {
```